things/redis/channels_test.go

```diff
@@ -0,0 +1,85 @@
+//
+// Copyright (c) 2018
+// Mainflux
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+package redis_test
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/mainflux/mainflux/things/redis"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestConnect(t *testing.T) {
+	channelCache := redis.NewChannelCache(cacheClient)
+
+	cid := uint64(123)
+	tid := uint64(321)
+
+	err := channelCache.Connect(cid, tid)
+	assert.Nil(t, err, fmt.Sprintf("connect new thing to channel: expected no error got %s\n", err))
+}
+
+func TestHasThing(t *testing.T) {
+	channelCache := redis.NewChannelCache(cacheClient)
+
+	cid := uint64(123)
+	tid := uint64(321)
+
+	channelCache.Connect(cid, tid)
+
+	cases := []struct {
+		desc      string
+		cid       uint64
+		tid       uint64
+		hasAccess bool
+	}{
+		{desc: "access check for thing that has access", cid: cid, tid: tid, hasAccess: true},
+		{desc: "access check for thing without access", cid: cid, tid: cid, hasAccess: false},
+		{desc: "access check for non-existing channel", cid: tid, tid: tid, hasAccess: false},
+	}
+
+	for _, tc := range cases {
+		hasAccess := channelCache.HasThing(tc.cid, tc.tid)
+		assert.Equal(t, tc.hasAccess, hasAccess, fmt.Sprintf("%s: expected %t got %t\n", tc.desc, tc.hasAccess, hasAccess))
+	}
+}
+func TestDisconnect(t *testing.T) {
+	channelCache := redis.NewChannelCache(cacheClient)
+
+	cid := uint64(123)
+	tid := uint64(321)
+
+	channelCache.Disconnect(cid, tid)
```

Comment:
It's nothing important, but it would be nice to use a table with test cases similar to what you did in [TestThingsID](https://github.com/mainflux/mainflux/blob/786b24d235f8bd948e0318e9f6e502441c1e8c99/things/redis/things_test.go#L38).

Reply:
I didn't use looping over a slice of test cases here because I wanted to call `Connect` or `Disconnect` before every test case

Reply:
I agree with @dusanb94. Connect one pair and test disconnect on that pair and some other pair that is not connected.

Reply:
Also, handle error.

Reply:
There's no handling errors because you won't get error when you disconect non-existing thing or non-existing channel, or removing non-existing channel.

Reply:
Still, you should check this error (put `require.Nil(t, err, ...)`), because behavior can change. Purpose of test is to validate that behavior.